### PR TITLE
[server] Implement EntitlementServiceUBP.hasPaidSubscription

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

- [x] Implement `hasPaidSubscription` logic in `EntitlementServiceUBP`.

Drive-by fix:
- [x] No longer "force" "more resources" in usage-based billing -- users can choose their resources, so we can deprecate `userGetsMoreResources` with usage-based billing (but it's still used by Chargebee customers)

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/12169

## How to test

1. Create a team called "Gitpod Something" to **enable the Usage-Based flag** for your user and team
2. Don't add a credit card just yet
3. ~~Try starting multiple workspaces in parallel -- at **4 parallel workspaces**, you should start seeing a paywall that prevents you from starting more (until the currently-running workspaces are stopped again)~~
4. ~~Once a workspace is running, you should **not be able to extend** the workspace timeout (i.e. no "clock" button in the IDE bottom bar)~~
5. Now, upgrade your team to paid by adding a credit card
6. Once upgraded, you should now be able to start up to **16 parallel workspaces** before seeing a paywall
7. You should also **be able to extend** the workspace timeout (using the "clock" button in the IDE bottom bar)

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview
- [x] /werft with-payment

### CODE REVIEW

File: components/server/ee/src/billing/entitlement-service-ubp.ts
Comment (geropl) [inline components/server/ee/src/billing/entitlement-service-ubp.ts:132]: :thought_balloon: Can we do this the other way around: check personal first, and then go to teams? The motivation is to try to reduce the number of checks required.

:cloud: Another idea: This impl checks all teams, where we could try to return after the first positive already. Maybe there is a simple way to get there? :thinking:

Comment (jankeromnes) [reply components/server/ee/src/billing/entitlement-service-ubp.ts:119]: Good ideas! 👍 Done and done. ✅

**Comment by jankeromnes** [comment]
Hmm, I thought we'd be able to test the Usage-Based free tier, but since it's not implemented, we can't:

Screenshot 2022-08-30 at 09 18 04

Also, unrelatedly, while opening several workspaces, I got a few of these error pages too. Looks like this retryAfter mechanism isn't properly supported in `CreateWorkspace`:

Screenshot 2022-08-30 at 09 19 35

**Comment by jankeromnes** [comment]
Hmm, I wasn't able to extend my workspace timeout (no clock button) 🤔 I guess this isn't quite ready yet & needs more debugging.

**Review by AlexTugarev — APPROVED** [review]
👍🏻  to unblock!

Maybe it would be good to simplify the promise races here ;-)

/hold

**Comment by jankeromnes** [comment]
Thanks! Good suggestion -- adjusted the code, and tested again, still works as intended. 🚀

/unblock

**Comment by geropl** [comment]
/unhold

I think that's what @jankeromnes wanted to do [here](https://github.com/gitpod-io/gitpod/pull/12305#issuecomment-1233931744).

**Pull request merged**
